content/faq/serverless.md

```diff
@@ -191,13 +191,27 @@ service: serverless-example
 plugins:
   - serverless-offline
 
+package:
+  patterns:
+    # here you can ignore some files or folders to reduce the bundle size.
+    - '!node_modules/**/LICENSE'
+    - '!node_modules/**/*.md'
+    - '!node_modules/**/*.js.map'
+    - '!node_modules/**/*.d.ts.map'
+    - '!node_modules/**/*.d.ts'
+    - '!node_modules/typeorm/browser/**'
+    - '!.webpack/**'
+    - '!src/**'
+    - '!test/**'
+    - '!.idea/**'
+
 provider:
   name: aws
   runtime: nodejs14.x
 
 functions:
   main:
-    handler: dist/main.handler
+    handler: dist/lambda.handler
```

Comment:
Why the change from `main`?

Reply:
Because now we export the configuration for lambda from `lambda.ts`. The name `main` or `lambda` is based on the file that we want to use to export handler to AWS.

Reply:
Right, but I think my question is more towards why are we not using `main.ts` for the lambda handler. Maybe I just didn't understand something elsewhere

Reply:
During API development in NestJS, we don't use `serverless framework`, we create an application with `createApp` and then call `app.listen`.

I could change the contents of `main.ts`, but that could lead to questions like bchilcott in #2142:

> However, using this in practice would require rebuilding every time a change is made, which is unrealistic and should be easily improved.

So to develop you can run: `npm run start:debug`, then to deploy to AWS Lambda, first test with `npx serverless offline` and then run `npx serverless deploy` to see the changes.

Think of deploying a NestJS API inside AWS Lambda not as deploying a resource or just a method, but the entire API, 10+ resources or something. It's not practical to build and then run `npx serverless offline` and explaining to the user to change commands like `npm run start:debug` just to integrate with AWS Lambda is more troublesome, instead just tell it to create another file.

Reply:
Although I understand the reasoning behind this change, I feel this is material for an article, not something that should be necessarily described in this chapter.